qemu/tests/qemu_guest_agent.py

```diff
@@ -905,27 +897,23 @@ class QemuGuestAgentBasicCheckWin(QemuGuestAgentBasicCheck):
     """
 
     @error_context.context_aware
-    def setup_gagent_in_host(self, session, params, vm):
-        error_context.context("download qemu-ga.msi to host", logging.info)
-        deps = params["deps"]
-        gagent_download_cmd = params["gagent_download_cmd"]
-        if deps == "yes":
-            deps_dir = data_dir.get_deps_dir("windows_ga_install")
-            gagent_download_cmd = gagent_download_cmd % deps_dir
-        utils.run(gagent_download_cmd,
-                  float(params.get("login_timeout", 360)))
-        gagent_host_path = params["gagent_host_path"]
-        if not os.path.exists(gagent_host_path):
-            raise error.TestFail("qemu-ga install program is not exist, maybe "
-                                 "the program is not successfully downloaded ")
-        gagent_guest_dir = params["gagent_guest_dir"]
-#        gagent_remove_service_cmd = params["gagent_remove_service_cmd"]
-        s, o = session.cmd_status_output("mkdir %s" % gagent_guest_dir)
-        if bool(s) and str(s) != "1":
-            raise error.TestError("Could not create qemu-ga directory in "
-                                  "VM '%s', detail: '%s'" % (vm.name, o))
-        error_context.context("Copy qemu guest agent program to guest", logging.info)
-        vm.copy_files_to(gagent_host_path, gagent_guest_dir)
+    def get_qga_pkg_path(self, session, params, vm):
+        """
+        Get the qemu-ga pkg path which would be installed.
+
+        :param session: VM session.
+        :param params: Dictionary with the test parameters
+        :param vm: Virtual machine object.
+        :return qemu_ga_pkg_path: Return the guest agent pkg path.
+        """
+        error_context.context("Get qemu-ga.msi path.", logging.info)
+        qemu_ga_pkg = params["qemu_ga_pkg"]
+        vol_virtio_key = "VolumeName like '%virtio-win%'"
+        vol_virtio = utils_misc.get_win_disk_vol(session, vol_virtio_key)
+        qemu_ga_pkg_path = r"%s:\%s\%s" % (vol_virtio, "guest-agent", qemu_ga_pkg)
```

Comment:
Please use `os.path.join()`
eg: `os.path.join("c:/", "sourcedir")`

Reply:
as os.path.join is running in linux host,so the result is not suitable for windows guest.